core/metacling/src/TCling.cxx

```diff
@@ -476,6 +476,10 @@ void TCling::UpdateEnumConstants(TEnum* enumObj, TClass* cl) const {
 
 TEnum* TCling::CreateEnum(void *VD, TClass *cl) const
 {
+#ifdef R__USE_CXXMODULES
+   cling::Interpreter::PushTransactionRAII RAII(fInterpreter);
```

Comment:
It would be nice to have a motivation for this in the commit log, especially why this is only needed for `R__USE_CXXMODULES`.